js/controllers.js

```diff
@@ -1,12 +1,192 @@
-var chartControllers = angular.module('chartControllers', []);
-
-chartControllers.controller('contactCtrl', ['$scope', '$http',
-  function ($scope, $http) {    
-      $scope.pageName = "contact";
-
-  }]);
-
-chartControllers.controller('chartCtrl', ['$scope', '$routeParams',
-  function($scope, $routeParams) {
-    $scope.pageName = "chartControllers";
-  }]);
\ No newline at end of file
+var chartControllers = angular.module('chartControllers', ['googlechart', 'nvd3', 'highcharts-ng']);
+
+
+chartControllers.controller('chartCtrl', ['$scope', '$location', 'Datax',
+    function($scope, $location, Datax) {
+        
+
+
+
+        //csv file data is load for all charts libraries
+        $scope.loadData = function(fileName) {   
+            //console.log("loading data "+fileName);
+            data = getData(fileName);
+            $scope.setActive('b1',fileName);
+            //$scope.Datax = Datax;
+            console.log(Datax)
+            loadYAxisLabel(fileName);
+            $scope.a = true;
+            drawChart();
+        };
+
+                //default one 
+        var myChart = {};
+       
+        $scope.loadChartLibrary = function(libraryIndex) {   
+            myChart = libraryName[libraryIndex];            
+            switchChartLib(myChart.library);
+            $scope.setActive('b2',myChart.library);
+            $scope.b = true;
+            drawChart();
+            
+        };
+ 
+
+        //getDataing highchart sample 
+        $scope.chartConfig = highchart.chartConfig;
+
+        //getting nvd3 chart sample
+        $scope.options = ncd3Chart1.options;
+        $scope.data = ncd3Chart1.data;
+
+        //getting google chart sample
+        $scope.chart = googleChart;
+
+        //Switch all chart libaraies chart types at one
+        $scope.switchChartType = function(gchartType, hchartType, dchartType) {
+            $scope.chart.type = gchartType;
+            $scope.chartConfig.options.chart.type = hchartType;
+            $scope.options.chart.type = dchartType;

```

Comment:
lines 47-49 look weird.
why do you keep chart types in so many places in your $scope?